bpf/pca.h

```diff
@@ -0,0 +1,96 @@
+#ifndef __PCA_H__
+#define __PCA_H__
+
+#include "utils.h"
+#include <string.h>
+
+static int attach_packet_payload(void *data, void *data_end, struct __sk_buff *skb, u64 current_time ){
+        payload_meta meta;
+        u64 flags = BPF_F_CURRENT_CPU;
+        // Enable the flag to add packet header
+        // Packet payload follows immediately after the meta struct
+        u32 packetSize = (__u32)(data_end-data);
+        // For packets which are allocated non-linearly struct __sk_buff does not necessarily 
+        // has all data lined up in memory but instead can be part of scatter gather lists. 
+        // This command pulls data from the buffer but incurs data copying penalty.
+        if (packetSize < skb->len){
+            packetSize = skb->len;
+            bpf_skb_pull_data(skb, skb->len);
```

Comment:
@shach33 can you add a check here if this return any error or not, it return 0 on success and else error.